include/envoy/http/header_map.h

```diff
@@ -659,6 +660,13 @@ class HeaderMap {
     headers.dumpState(os);
     return os;
   }
+
+  /**
+   * Return the optional formatter attached to this header map. Filters can use the non-const
+   * version to remember additional header keys during operation if they wish.
+   */
+  virtual HeaderKeyFormatterOptConstRef formatter() const PURE;
+  virtual StatefulHeaderKeyFormatterOptRef formatter() PURE;
```

Comment:
it is surprising to have two overloaded methods with no args, each of which return types that differ by something other than const-ness.

Reply:
Sure I can rename.